packages/runtime-common/indexer/client.ts

```diff
@@ -41,9 +63,31 @@ interface GetEntryOptions {
   useWorkInProgressIndex?: boolean;
 }
 
+interface QueryResultsMeta {
+  // TODO SQLite doesn't let us use cursors in the classic sense so we need to
```

Comment:
What do you mean by cursors in the classic sense? I am familiar with implementing cursor based pagination where the last value of the column from the previous query is selected as the cursor for the next set. This sounds like it would be achievable with SQLite also. Just wondering what kind of pagination you had in mind here

Reply:
SQLite does have a cursor available, but it’s only available from the synchronous API. 

Normally you get it like this: 
```
import sqlite3;
con = sqlite3.connect("tutorial.db");
con.cursor()cur = con.cursor();
result = cur.execute("SELECT name FROM sqlite_master");
result.fetchone()

```

however from the async interface you no longer have a direct handle on the connection. Rather the connection is only available in the worker thread since it’s not possible for the connection to be serialized over the postMessage boundary. 

The worker thread only has a couple API calls:
close(), config-get(), execute(), open(). https://sqlite.org/wasm/doc/trunk/api-worker1.md
the idea being only POJO’s are available to cross the postMessage boundary between a worker thread and the main thread

Reply:
So instead we’ll have to manually control the result set as I described in this comment. We’ll need to use the realm version so that we can prevent the index mutations made between pages from effecting the results (so the realm version will be part of the query), as well as only fetching one page at a time using the ROW_NUMBER() and/or LIMIT sql commands by having the caller keep track of what page number of results they are requesting and including that in our `Query` interface.

Reply:
Ah! I see now what you mean by the cursor and the boundary.